drivers/sensor/max17262/max17262.c

```diff
@@ -0,0 +1,342 @@
+		 * init and continue with measurements.
+		 */
+		LOG_DBG("No POR event detected - skip device configuration");
+		return 0;
+	}
+	LOG_DBG("POR detected, setting custom device configuration...");
+
+	/** STEP 1 */
+	max17262_reg_read(dev, FSTAT, &tmp);
+
+	/* Do not continue until FSTAT.DNR bit is cleared */
+	while (tmp & FSTAT_DNR)
+		max17262_reg_read(dev, FSTAT, &tmp);
+
+	/** STEP 2 */
+	/* Store original HibCFG value */
+	max17262_reg_read(dev, HIBCFG, &hibcfg);
+
+	/* Exit Hibernate Mode step 1 */
+	max17262_reg_write(dev, SOFT_WAKEUP, 0x0090);
+	/* Exit Hibernate Mode step 2 */
+	max17262_reg_write(dev, HIBCFG, 0x0000);
+	/* Exit Hibernate Mode step 3 */
+	max17262_reg_write(dev, SOFT_WAKEUP, 0x0000);
+
+	/** STEP 2.1 --> OPTION 1 EZ Config (No INI file is needed) */
+	/* Write DesignCap */
+	max17262_reg_write(dev, DESIGN_CAP, config->design_cap);
+
+	/* Write IChgTerm */
+	max17262_reg_write(dev, ICHG_TERM, config->desired_charging_current);
+
+	/* Write VEmpty */
+	max17262_reg_write(dev, VEMPTY, ((config->empty_voltage / 10) << 7) |
+					  ((config->recovery_voltage / 40) & 0x7F));
+
+	/* Write ModelCFG */
+	if (config->charge_voltage > 4275)
+		max17262_reg_write(dev, MODELCFG, 0x8400);
+	else
+		max17262_reg_write(dev, MODELCFG, 0x8000);
+
+	/*
+	 * Read ModelCFG.Refresh (highest bit),
+	 * proceed to Step 3 when ModelCFG.Refresh == 0
+	 */
+	max17262_reg_read(dev, MODELCFG, &tmp);
+
+	/* Do not continue until ModelCFG.Refresh == 0 */
+	while (tmp & MODELCFG_REFRESH)
+		max17262_reg_read(dev, MODELCFG, &tmp);
+
+	/* Restore Original HibCFG value */
+	max17262_reg_write(dev, HIBCFG, hibcfg);
+
+	/** STEP 3 */
+	/* Read Status register */
+	max17262_reg_read(dev, STATUS, &tmp);
+
+	/* Clear PowerOnReset bit */
+	tmp &= ~STATUS_POR;
+	max17262_reg_write(dev, STATUS, tmp);
+
+	return 0;
+}
+
+static const struct sensor_driver_api max17262_battery_driver_api = {
+	.sample_fetch = max17262_sample_fetch,
+	.channel_get = max17262_channel_get,
+};
+
+#define MAX17262_INIT(n)						\
+	static struct max17262_data max17262_driver_##n;		\
+									\
+	static const struct max17262_config max17262_config_##n = { \
+		.i2c_bus = DT_INST_BUS_LABEL(n),			\
+		.i2c_addr = DT_INST_REG_ADDR(n),			\
+		.design_voltage = DT_INST_PROP(n, design_voltage),	\
+		.desired_voltage = DT_INST_PROP(n, desired_voltage), \
+		.desired_charging_current =				\
+			DT_INST_PROP(n, desired_charging_current),	\
+		.design_cap = DT_INST_PROP(n, design_cap), \
+		.empty_voltage = DT_INST_PROP(n, empty_voltage), \
+		.recovery_voltage = DT_INST_PROP(n, recovery_voltage), \
+		.charge_voltage = DT_INST_PROP(n, charge_voltage), \
+	};								\
+									\
+	DEVICE_DT_INST_DEFINE(n, &max17262_gauge_init,		\
+			    device_pm_control_nop,			\
+			    &max17262_driver_##n,			\
+			    &max17262_config_##n, POST_KERNEL,	\
+			    CONFIG_SENSOR_INIT_PRIORITY,		\
+			    &max17262_battery_driver_api)
+
+DT_INST_FOREACH_STATUS_OKAY(MAX17262_INIT);
```

Comment:
This won't work if there are multiple instances defined. It should be:

```suggestion
			    &max17262_battery_driver_api);

DT_INST_FOREACH_STATUS_OKAY(MAX17262_INIT)
```

The `MAX17262_INIT` macro is responsible for its own separators, like `;`.

To see for yourself, build the sample with an app.overlay that has two devices, like this:

```
&arduino_i2c {
	status = "okay";

	max17262@36 {
		compatible = "maxim,max17262";
		reg = <0x36>;
[...]
	};

	max17262@37 {
		compatible = "maxim,max17262";
		reg = <0x37>;
[...]
	};
};

```

(The 0x37 address may not be realistic, but hopefully you get my meaning.)

Reply:
Implemented, it works now with multiple DT nodes.